include/boost/geometry/util/math.hpp

```diff
@@ -525,6 +525,30 @@ struct rounding_cast<Result, Source, true, false>
     }
 };
 
+template <typename T, bool IsIntegral = std::is_integral<T>::value>
+struct divide
+{
+    static inline T apply(T const& n, T const& d)
+    {
+        return n / d;
+    }
+};
+
+template <typename T>
+struct divide<T, true>
+{
+    static inline T apply(T const& n, T const& d)
+    {
+        return n == 0 ? 0
+          : n < 0
+          ? (d < 0 ? (n + (-d + 1) / 2) / d + 1
+                   : (n + ( d + 1) / 2) / d - 1  )
+          : (d < 0 ? (n - (-d + 1) / 2) / d - 1
+                   : (n - ( d + 1) / 2) / d + 1  )
```

Comment:
More interesting is what would happen here if d was 0. The answer is the same as above, i.e.: the same as before this change was made, i.e. undefined behavior.

Reply:
I thought about it too but since the integer division lines up with the floating point division, the behavior should be the same. In other words, if we check integer divisor 0, we should also check floating point divisor 0...